packages/babel-plugin-proposal-decorators/test/fixtures/2023-05-misc--to-es2015/valid-expression-formats/output.js

```diff
@@ -1,11 +1,12 @@
-var _initProto, _initClass, _classDecs, _dec, _dec2, _dec3, _obj, _dec4, _Foo2;
+var _initProto, _initClass, _obj, _classDecs, _dec, _dec2, _dec3, _obj2, _dec4, _Foo2;
 const dec = () => {};
-_classDecs = [void 0, dec, void 0, call(), void 0, chain.expr(), void 0, arbitrary + expr, array, array[expr]];
+_obj = array;
```

Comment:
Here if ,`array` was a local variable we wouldn't memoize it, right?

Reply:
Right

Reply:
Yes, we will memoize it when it is a `const` reference

https://github.com/babel/babel/blob/bfefe7001332ee000364a210b26afd822d17e97f/packages/babel-traverse/src/scope/index.ts#L557-L572

I think the `pureGetters` could extend to global variables as well, so when this assumption is `true`, we don't memoise global variables as well as member expressions.